alacritty_terminal/src/grid/mod.rs

```diff
@@ -645,19 +640,25 @@ impl<T> Grid<T> {
     }
 
     /// This is used only for initializing after loading ref-tests
+    #[inline]
     pub fn initialize_all(&mut self, template: &T)
     where
         T: Copy + GridCell,
     {
-        let history_size = self.raw.len().saturating_sub(*self.lines);
-        self.raw.initialize(self.max_scroll_limit - history_size, template, self.cols);
+        // Remove all cached lines to clear them of any content
+        self.truncate();
+
+        // Initialize everything with empty new lines
+        self.raw.initialize(self.max_scroll_limit - self.history_size(), template, self.cols);
     }
 
     /// This is used only for truncating before saving ref-tests
```

Comment:
I don't believe this comment holds true anymore, now that we're using it in `initialize_all` above.

Reply:
This is a new comment I've added in this PR. I'm not sure what you mean?

Reply:
Brainfart. Move along, nothing to see here.